feat(integrations/teams)!: implement standard markdown support

JustinBordage opened the pull request.

JustinBordage committed:
feat: implement markdown-to-teams-xml

JustinBordage committed:
fix: test descriptions that were lingering from the telegram tests

JustinBordage committed:
chore: bump botbuilder version

JustinBordage committed:
refactor: adjusted some of the unit tests to better reflect e2e qa testing

JustinBordage committed:
feat: adjust html sanitizer rules & implement option override in "sanitizeHtml" function

JustinBordage committed:
feat: implement logic to strip anchor/hyperlink tags without href attributes

JustinBordage committed:
feat: add "turndown" package for parsing html into markdown

JustinBordage committed:
feat: implement html-to-markdown converter & accompanying unit tests

JustinBordage committed:
style: rename markdown transform functions

- Renamed: "teamsHtmlToStdMarkdown" to "transformTeamsHtmlToStdMarkdown"
- Renamed: "transformMarkdown" to "transformMarkdownToTeamsXml"

JustinBordage committed:
style: Rename "Card" type in channels/outbound.ts to "BotpressCard"

JustinBordage committed:
feat: implement markdown to xml transformer in the markdown message channel type

JustinBordage committed:
feat!: move markdown conversion logic from markdown channel to text channel & remove markdown channel

JustinBordage committed:
feat: implement html-to-markdown conversion for inbound channel messages

JustinBordage committed:
style: cleanup stale comments

JustinBordage committed:
docs: fix comments in unit tests

mikee-b commented on integrations/teams/src/channels/inbound.ts:
```diff
@@ -46,15 +47,32 @@ export const processInboundChannelMessage = async ({ client, ctx, logger }: bp.H
 }
+
+/** Attempts to extract & convert the rich text if it can find it, otherwise it falls back to plain text */
+const _attemptRichTextExtractionAndConversion = (activity: Activity): string => {
```
**nitpick:** I feel like we generally prefix functions like this with `try` rather than `attempt`.

Math-Fauch reviewed (approved):
This looks very well done to me, wait on fleur's approval but everything seems well-tested and safer than what is in common.

JustinBordage committed:
fix: remove unnecessary oxlint disable

JustinBordage committed:
Merge branch 'jbord_teams-v2.0.0' into jbord_teams-implement-standard-markdown-support

JustinBordage committed:
fix: incorrect type name from merge

JustinBordage committed:
Merge branch 'jbord_teams-v2.0.0' into jbord_teams-implement-standard-markdown-support

JustinBordage committed:
fix: message extraction following merge

franklevasseur commented on integrations/teams/src/markdown/markdown-to-teams-xml.ts:
```diff
@@ -0,0 +1,220 @@
+  TableCellWithHeaderInfo,
+} from './types'
+
+export const stripAllHandlers: MarkdownHandlers = {
```
**chore:** This looks a lot like what is available in `common`, have you looked into sharing some code?

franklevasseur reviewed (approved):
Good Work! I kind of expected more code to be shared with `common`... Let's talk tomorrow so I can understand if it's doable

JustinBordage committed:
refactor: move "tests/types.ts" to "src/types.ts"

- Also removed "tests/**/*" from the "ts-config.json" since it's no longer required

JustinBordage committed:
feat: add tests for nested lists in "markdown-to-teams-xml.test.ts"

JustinBordage committed:
style: fix naming convention of functions

JustinBordage committed:
refactor: minor adjustments to names & comments

- rename "stripAllHandlers" to "defaultHandlers"
- remove unused parameters from "defaultHandlers"
- fix comment (Remove "Remark Textr" since it's not being used)
